core/frontend/src/tile/Tile.ts

```diff
@@ -418,7 +446,8 @@ export abstract class Tile {
         return TileVisibility.Visible;
     }
 
-    const pixelSize = args.getPixelSize(this);
+    const scale = this.computePixelSizeScaleFactor(args);

```

Comment:
Can you avoid computing this for every tile by, e.g., computing in TileDrawArgs constructor and storing?

Reply:
Yea, I was wondering if I should do something like that.

Reply:
done